pkg/cache/snapshot_test.go

```diff
@@ -45,394 +43,450 @@ func TestSnapshot(t *testing.T) {
+						UsedResources:        ResourceQuantities{},
+						Workloads:            map[string]*workload.Info{},
+						Preemption:           defaultPreemption,
 					},
-					{
-						Name: "example.com/gpu",
-						Flavors: []kueue.Flavor{
-							{
-								Name: "default",
-								Quota: kueue.Quota{
-									Min: resource.MustParse("50"),
-								},
-							},
-						},
+					"b": {
+						Name:                 "b",
+						NamespaceSelector:    labels.Everything(),
+						Status:               active,
+						RequestableResources: map[corev1.ResourceName]*Resource{},
+						UsedResources:        ResourceQuantities{},
+						Workloads:            map[string]*workload.Info{},
+						Preemption:           defaultPreemption,
 					},
 				},
+				ResourceFlavors: map[string]*kueue.ResourceFlavor{},
 			},
 		},
-		{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "flavor-nonexistent-cq",
-			},
-			Spec: kueue.ClusterQueueSpec{
-				Cohort: "foo",
-				Resources: []kueue.Resource{
-					{
+		"inactive clusterQueues": {
+			cqs: []*kueue.ClusterQueue{
+				utiltesting.MakeClusterQueue("flavor-nonexistent-cq").
+					Resource(&kueue.Resource{
 						Name: corev1.ResourceCPU,
 						Flavors: []kueue.Flavor{
-							{
-								Name: "nonexistent-flavor",
-								Quota: kueue.Quota{
-									Min: resource.MustParse("100"),
-								},
-							},
+							*utiltesting.MakeFlavor("nonexistent-flavor", "100").Obj(),
 						},
-					},
-				},
+					}).Obj(),
+			},
+			wantSnapshot: Snapshot{
+				ClusterQueues:            map[string]*ClusterQueue{},
+				ResourceFlavors:          map[string]*kueue.ResourceFlavor{},
+				InactiveClusterQueueSets: sets.New("flavor-nonexistent-cq"),
 			},
 		},
-		{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "bar",
+		"resourceFlavors": {
+			rfs: []*kueue.ResourceFlavor{
+				utiltesting.MakeResourceFlavor("demand").
+					MultiLabels(map[string]string{"a": "b", "instance": "demand"}).Obj(),
+				utiltesting.MakeResourceFlavor("spot").
+					MultiLabels(map[string]string{"c": "d", "instance": "spot"}).Obj(),
+				utiltesting.MakeResourceFlavor("default").Obj(),
 			},
-			Spec: kueue.ClusterQueueSpec{
-				Resources: []kueue.Resource{
-					{
-						Name: corev1.ResourceCPU,
-						Flavors: []kueue.Flavor{
-							{
-								Name: "default",
-								Quota: kueue.Quota{
-									Min: resource.MustParse("100"),
-								},
-							},
-						},
-					},
+			wantSnapshot: Snapshot{
+				ClusterQueues: map[string]*ClusterQueue{},
+				ResourceFlavors: map[string]*kueue.ResourceFlavor{
+					"demand": utiltesting.MakeResourceFlavor("demand").
+						MultiLabels(map[string]string{"a": "b", "instance": "demand"}).Obj(),
+					"spot": utiltesting.MakeResourceFlavor("spot").
+						MultiLabels(map[string]string{"c": "d", "instance": "spot"}).Obj(),
+					"default": utiltesting.MakeResourceFlavor("default").Obj(),
 				},
 			},
 		},
-		{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "with-preemption",
+		"workloads": {
```

Comment:
I would just merge this one with the test case above titled "clusterQueues"

Reply:
I'm ok with that.

Reply:
Done.